core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -844,8 +845,8 @@ private[deploy] class Master(
       // We assign workers to each waiting driver in a round-robin fashion. For each driver, we
       // start from the last worker that was assigned a driver, and continue onwards until we have
       // explored all alive workers.
-      var launched = false
-      var isClusterIdle = true
+      var launched = (drivers.size - waitingDrivers.size) >= maxDrivers
```

Comment:
Hmm, isn't this making submitting driver silently ignored? For example, there is a warning log for the case the driver needs more resources than any worker. Should we also have a warning log for the case that drivers reaches the maximum drivers?

Reply:
No~ `schedule` is re-trigger in many cases and especially on the removal of running drivers. So, if the number of running drivers dropped down, new drivers will be allocated to the slots, @viirya .

Reply:
Got it. I just worry about the case that submitted drivers are silently waiting there without any feedback to users.